### PR TITLE
Fix NPE in the TO logs during replication change

### PR DESCRIPTION
**Pull request by stanlyDoge**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Bugfix

### Description
When the topic's replication is changed, the NPE is printed in the TO logs. This should remove the NPE but does not fix the problem.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by tombentley** [comment]
> does not fix the problem

You should clarify what "the problem" is in this context. It doesn't fix the fact that the TO does not support changes to replication factor.

File: topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java [hunk 1]
```diff
@@ -308,7 +306,7 @@ private String generateReassignment(Topic topic, String zookeeper) throws IOExce
         final ProcessHelper.ProcessResult processResult = ProcessHelper.executeSubprocess(executeArgs);
         ProcessHelper.delete(topicsToMove);
         String json = forEachLineStdout(processResult, new ReassignmentLineParser());
-        return json;
+        return json == null ? "{\"error\": \"Reassign Partitions not implemented\"}" : json;
```
Comment (ppatierno) [inline topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java]: I would say "not supported".

File: topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java [hunk 2]
```diff
@@ -308,7 +308,7 @@ private String generateReassignment(Topic topic, String zookeeper) throws IOExce
         final ProcessHelper.ProcessResult processResult = ProcessHelper.executeSubprocess(executeArgs);
         ProcessHelper.delete(topicsToMove);
         String json = forEachLineStdout(processResult, new ReassignmentLineParser());
-        return json;
+        return json == null ? "{\"error\": \"Reassign Partitions is not supported\"}" : json;
```
Comment (tombentley) [inline topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java]: Why not just return a failed Future from `OperatorAssignedKafkaImpl.changeReplicationFactor(Topic topic)`?

File: topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java [hunk 3]
```diff
@@ -87,6 +81,9 @@ public OperatorAssignedKafkaImpl(AdminClient adminClient, Vertx vertx, Config co
 
     @Override
     public Future<Void> changeReplicationFactor(Topic topic) {
+
+        return Future.failedFuture("Changing replication factor is not supported");
+        /*
```
Comment (scholzj) [inline topic-operator/src/main/java/io/strimzi/operator/topic/OperatorAssignedKafkaImpl.java]: What is the point of this huge comment? Do we plan to use this code later? Or sould it be just removed?

**Comment by tombentley** [comment]
Fixed by https://github.com/strimzi/strimzi-kafka-operator/pull/1974

**Pull request closed without merging**